jwst/outlier_detection/spec.py

```diff
@@ -93,7 +93,15 @@ def detect_outliers(
 
     # Perform median combination on set of drizzled mosaics
     # create_median should be called as a method from parent class
-    median_data = create_median(drizzled_models, maskpt)
+    if in_memory:
+        buffer_size = None
+    else:
+        # reasonable buffer size is the size of an input model, since that is
+        # the smallest theoretical memory footprint
```

Comment:
I'm wondering where this limit came from.

I think since the drizzled models are in memory (for in_memory=True) we will have to have at least 1 drizzled model in memory. We'll also always have all the input models in memory. So during generation of the 1st drizzled models we will have some baseline memory usage + all input models + 1 drizzled model. If we split up the drizzled model and write it out to many files (what this PR does, right? I haven't finished looking at it...) we can throw out the drizzled model before generating the next one (not what this PR does but what we might work towards). So our peak memory usage will be: baseline + n input models + 1 drizzled models. After drizzling we can go back down to baseline + n input models. We will have to then create a median image (equal to 1 drizzled array, but not model since we don't need weights etc) and then load each temporary file and fill in the median image. I think that gives a peak memory usage of: baseline + n input models + median image. I think this is less than baseline + n input models + 1 drizzled models by at least the weight array (the same size as the drizzled data). That's a very long winded way of saying, can't this be the size of the drizzled array and not the input array? If so, I think it would simplify the API as the buffer size could be computed within create_median (as it's written now).

Reply:
I agree with you that we can use the drizzled array size, so I will change that.

I'm almost certainly missing something, but why do all the input models need to be in memory?  Do you mean within `resample.do_drizzle` or somewhere else?

Reply:
You're right. My Friday brain was conflating this with the in memory case. Ignore the bit about the n input models.

Reply:
The [default buffer size computation](https://github.com/emolter/jwst/blob/1fdfee4deada2664110351794d40277e13941ce8/jwst/outlier_detection/utils.py#L220) should be scaled by the size of the resampled images now, let me know if it looks ok to you